nativelib/src/main/scala/scala/scalanative/unsafe/package.scala

```diff
@@ -238,7 +238,6 @@ package object unsafe {
         val $size   = _root_.scala.scalanative.unsafe.sizeof[$T]($tag)
         val $ptr    = $z.alloc($size)
         val $rawptr = $runtime.toRawPtr($ptr)
```

Comment:
Again, I will discuss this in an Issue.

To express my concern in short:

In full generality, one would have both an allocator for cleared memory (this PR edit) and one
for raw memory (before this PR).  I suspect that almost all cases, memory allocated on the
stack is going to be immediately overwritten by other data. That is, the clearing is wasted.

A developer knows if they want cleared or raw memory and could/should call the proper
method.  The key here is that the method have a name which gives a clue and documentation
which says exactly what it does.

Reply:
To try to describe my thoughts.  If there is only one stack allocator (pair?) it should have the existing behavior
to minimize change.  It is documented in package.scala as not clearing memory and there is no `Zone`
equivalent to have opposite behavior.

The question, at least in my mind, if implementing a new  `stackallocZeroed` or `stackallocCleared` is
worth the added complexity.  How one answers that  may be a matter of personal programming style, expectation,
and experience.

Reply:
Perhaps the stackalloc changes could be split out to another so that the `z.alloc` changes can advance quickly?

Reply:
`stackalloc` is tricky one. Tehcnically it is `{0}` and for LLVM that I do have and investigate it adds `memset`.

Anyway, I haven't got time proof that it is behaviour for all LLVM/clang and not some side effect for macOS.

As soon as I do have some time for SN I'll do this investiagetion and if it is true I'll remove this `memset` but I don't like blind shot without proof on this case because it is easy to change and very difficult to track.